lib/report_portal/cucumber/parallel_report.rb

```diff
@@ -35,65 +35,65 @@ def initialize(desired_time)
         ReportPortal.last_used_time = 0
         set_parallel_tests_vars
         if ParallelTests.first_process?
-          File.open(file_with_launch_id, 'w') do |f|
-            f.flock(File::LOCK_EX)
-            start_launch(desired_time, @cmd_args_of_parallel_tests)
-            f.write(ReportPortal.launch_id)
-            f.flush
-            f.flock(File::LOCK_UN)
-          end
-          @folder_creation_tracking_file = Pathname(Dir.tmpdir)  + "folder_creation_tracking_#{ReportPortal.launch_id}.lck"
-          File.open(@folder_creation_tracking_file, 'w+') do |f|
-            f.flock(File::LOCK_EX)
-            f.flush
-            f.flock(File::LOCK_UN)
-          end
+          start_launch(desired_time)
         else
           start_time = monotonic_time
           loop do
-            break if File.exist?(file_with_launch_id)
+            break if File.exist?(lock_file)
             if monotonic_time - start_time > wait_time_for_launch_start
               raise "File with launch ID wasn't created after waiting #{wait_time_for_launch_start} seconds"
             end
             sleep 0.5
           end
-          File.open(file_with_launch_id, 'r') do |f|
-            f.flock(File::LOCK_SH)
+          File.open(lock_file, 'r') do |f|
             ReportPortal.launch_id = f.read
-            f.flock(File::LOCK_UN)
           end
-          sleep_time = ENV['TEST_ENV_NUMBER'].to_i
+          add_process_description
+          sleep_time = 5
           sleep(sleep_time) # stagger start times for reporting to Report Portal to avoid collision
         end
       end
 
+      def add_process_description
```

Comment:
Could you explain the use case for updating the description?
If the launch description is static, it can be set by the process that starts the launch.

Reply:
When you execute `parallel_cucumber -o  ' features/*'  -p 4` assuming you have 4 feature files:
-> cucumber features/file1.feture
-> cucumber features/file2.feture
-> cucumber features/file3.feture
-> cucumber features/file4.feture

Since the description is parsed from command args, it will only have one feature file name. Therefore I added the option each process add its's own description.